proto/document/v1/document.proto

```diff
@@ -0,0 +1,126 @@
+syntax = "proto3";
+package nitric.document.v1;
+
+import "google/protobuf/struct.proto";
+
+// protoc plugin options for code generation
+option go_package = "nitric/v1;v1";
+option java_package = "io.nitric.proto.document.v1";
+option java_multiple_files = true;
+option java_outer_classname = "Documents";
+option php_namespace = "Nitric\\Proto\\Document\\V1";
+option csharp_namespace = "Nitric.Proto.Document.v1";
+
+// Service for storage and retrieval of simple JSON keyValue
+service Document {
+  // Get an existing document
+  rpc Get (DocumentGetRequest) returns (DocumentGetResponse);
+
+  // Create a new or overwrite an existing document
+  rpc Set (DocumentSetRequest) returns (DocumentSetResponse);
+
+  // Delete an existing document
+  rpc Delete (DocumentDeleteRequest) returns (DocumentDeleteResponse);
+  
+  // Query the document collection (pagination support)
+  rpc Query (DocumentQueryRequest) returns (DocumentQueryResponse);
+  
+  // Query All the document collection (streaming support)
+  rpc QueryAll (DocumentQueryAllRequest) returns (stream DocumentQueryAllResponse);
+}
+
+message Key {
+  // The item collection
+  string collection = 1;
+
+  // The items unique id
+  string id = 2;
+}
+
+message DocumentGetRequest {
+  // The document root key
+  Key key = 1;
+
+  // The optional sub collection key
+  Key sub_key = 2;
+}
+
+message DocumentGetResponse {
```

Comment:
I think we should have a `Document` message type, which contains the value (and in future could contain metadata like createdTime or whatever if needed)

```
message Document {
   google.protobuf.Struct content = 1;
}
```

A single 'Document' could be returned from a 'Get' request. A ResultPage (or similar) containing repeated Documents could be returned from Query and a steam of Documents could be returned from QueryStream.

Reply:
This would be similar to how the `Key` message type was extracted for reuse.

Reply:
Just want to confirm you mean something like this:
```
service DocumentService {
  // Get an existing document
  rpc Get (DocumentGetRequest) returns (DocumentGetResponse);

  // Create a new or overwrite an existing document
  rpc Set (DocumentSetRequest) returns (DocumentSetResponse);

  // Delete an existing document
  rpc Delete (DocumentDeleteRequest) returns (DocumentDeleteResponse);
  
  // Query the document collection (supports pagination)
  rpc Query (DocumentQueryRequest) returns (DocumentQueryResponse);
  
  // Query the document collection (supports streaming)
  rpc QueryStream (DocumentQueryStreamRequest) returns (stream Document);
}

message Document {
  // The document content (JSON object)
  repeated google.protobuf.Struct content = 2;
}
```

 Also did you did you mean return a Document message directly from the rpc: 
 ```
   // Get an existing document
  rpc Get (DocumentGetRequest) returns (Document);
 ```
 
Note with this change we should also call our services XxxService to avoid name collisions with the message types, and we should probably do this across all the proto contracts.